### PR TITLE
Pre-merge cleanup

### PR DESCRIPTION
**Pull request by hlinnaka**
Misc patches to reduce the diff vs. upstream, and do some other small fixes and cleanups. These were lifted from the merge_8_3_22 branch we've been working on. Let's push these to master before the big merge hits master, in order to keep the merge commit slightly smaller, and because they're not really related to the merge anyway.

### CODE REVIEW

File: src/include/postmaster/checkpoint.h [hunk 1]
```diff
@@ -1,27 +0,0 @@
...
-extern int	CheckPointWarning;
-
-extern void CheckpointMain(void);
-
-extern void RequestCheckpoint(bool waitforit, bool warnontime);
-
-extern Size CheckpointShmemSize(void);
-extern void CheckpointShmemInit(void);
-
-extern int checkpoint_start(void);
```
Comment (danielgustafsson) [inline src/include/postmaster/checkpoint.h]: As a note to other reviewers who might be wondering the same thing, `checkpoint_start()` doesn't exist anymore and thus this declaration wasn't pulled over to `bgwriter.h`.

Comment (hlinnaka) [inline src/include/postmaster/checkpoint.h]: Ah, I didn't actually notice that myself!

File: src/backend/parser/gram.y [hunk 2]
```diff
@@ -8001,28 +8001,28 @@ CreateConversionStmt:
 /*****************************************************************************
  *
  *		QUERY:
- *				CLUSTER [VERBOSE] <qualified_name> [ USING <index_name> ]
- *				CLUSTER [VERBOSE]
- *				CLUSTER [VERBOSE] <index_name> ON <qualified_name> (for pre-8.3)
```
Comment (danielgustafsson) [inline src/backend/parser/gram.y]: This syntax isn't listed in the documentation at all for 4.3 so it might even be worth bringing up.

**Comment by danielgustafsson** [comment]
:+1:

**Comment by hlinnaka** [comment]
Pushed, thanks!

**Pull request closed without merging**